applications/dashboard/controllers/class.entrycontroller.php

```diff
@@ -468,9 +468,15 @@ public function Connect($Method) {
             $RoleIDs = $this->UserModel->NewUserRoleIDs();
          }
 
+         if (C('Garden.SSO.SyncRolesBehavior') === 'register') {
+            $SaveRolesRegister = TRUE;
+            $SaveRoles = FALSE;
+         }
+
```

Comment:
It doesn't seem right that you can get out of this block without `$SaveRolesRegister` being set at all (if `Garden.SSO.SyncRoles` == `true` and `Garden.SSO.SyncRolesBehavior` == `false`).